cvmfs/catalog_mgr_rw.cc

```diff
@@ -837,6 +845,36 @@ shash::Any WritableCatalogManager::SnapshotCatalog(WritableCatalog *catalog)
   return hash_catalog;
 }
 
+void WritableCatalogManager::DoBalance() {
+  ReadLock();
+  const CatalogList &catalog_list = GetCatalogs();
+  for (unsigned i = 0; i < catalog_list.size(); ++i) {
```

Comment:
This traverses the catalog tree from top to bottom, right?  Isn't it supposed to traverse the catalog list from bottom to top?